app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -108,7 +109,15 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
     private var isTurboEnabled: Boolean
         get() = Settings.getInstance(context).isBlockingEnabled
         set(value) {
-            Settings.getInstance(context).isBlockingEnabled = value
+            val settings = Settings.getInstance(context)
+            settings.isBlockingEnabled = value
+
+            val engineSession = context.components.sessionManager.getOrCreateEngineSession()
+            if (value) {
+                engineSession.enableTrackingProtection(settings.trackingProtectionPolicy)
```

Comment:
What would happen if `settings.isBlockingEnabled` and `engineSession.trackingProtectionEnabled` went out of sync?  This looks like a chunk of code where bugs could pretty easily worm their way in.  Testing setters is usually overkill, but I think it might be worthwhile here just in case someone who is unfamiliar with this setup accidentally tweaks it.

More broadly, it seems like we're doing an awesome job of breaking out components, but if these two both need this information I wonder if they might benefit from some adjustment.  Are there any concerns that could be moved from one to another?  If not, would it be possible to place some of this behind a facade?